### PR TITLE
STRG-2286 - Use widget factory to select appropriate strigo widget

### PR DESCRIPTION
**Pull request by v0vs**
## Context

Get strigo widget from widget factory to consolidate the widget interface and avoid duplicating
same switch case widget selection code in multiple places.
Move widget types to a dedicated file.

## Tests

- [x] Strigo event sent.
- [x] Overlay and iframe widgets are displayed with proper loading.
- [x] Configuration is loaded after browser refresh

### CODE REVIEW

**Comment by nir0s** [comment]
@v0vs, do you want a review on this?

**Comment by v0vs** [comment]
@DvirMalka Gonna need your help with the configuration test - I'm not sure I follow the flow there.

File: src/modules/widgets/widget.types.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  OVERLAY = "overlay",
+  DYNAMIC = "dynamic"
+}
+
+export interface StrigoWidget {
+  init: () => SDK_TYPES;
+  setup: (params: { development: boolean, version?: string }) => void;
+  shutdown: () => void;
+  hideLoader: () => void;
+  initEventListeners: (academyPlayerFrame: HTMLIFrameElement) => void;
```
Comment (v0vs) [inline src/modules/widgets/widget.types.ts:14]: Make private

File: src/modules/widgets/widget.types.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  IFRAME = "iframe",
+  OVERLAY = "overlay",
+  DYNAMIC = "dynamic"
+}
+
+export interface StrigoWidget {
+  init: () => SDK_TYPES;
+  setup: (params: { development: boolean, version?: string }) => void;
+  shutdown: () => void;
+  hideLoader: () => void;
```
Comment (v0vs) [inline src/modules/widgets/widget.types.ts:13]: finishSetup

File: src/strigo/index.ts [hunk 3]
```diff
@@ -4,38 +4,31 @@ import * as urlTools from "../modules/url/url";
...
-        SDKType = SDK_TYPES.OVERLAY;
-        overlayWidget.init();
-        break;
-      }
-      default:
-        break;
+    const widget = widgetFactory.getWidget(sessionManager.getWidgetFlavor());
+
+    if (!widget) {
+      Logger.error(`Can't finish initialization. widgetFlavor is not supported.`);
```
Comment (v0vs) [inline src/strigo/index.ts:26]: Possibly throw an error in getWidget function instead of logging the error

**Comment by v0vs** [comment]
@DvirMalka @nir0s I've made the changes. Again, not optimal - for example, I think we should straighten the setup flow including the academy player iframe loaded flow using promises, but there are a lot of changes and it's risky.

**Review by DvirMalka — APPROVED** [review]

**Comment by DvirMalka** [comment]
LTGM and add the extra tweaks later

**Review by nir0s — APPROVED** [review]

**Pull request merged**